pkg/auth/handler/webapp/csrf_middleware.go

```diff
@@ -36,9 +36,13 @@ type CSRFMiddleware struct {
 	BaseViewModel           *viewmodels.BaseViewModeler
 	Renderer                Renderer
 	UIImplementationService CSRFMiddlewareUIImplementationService
+	EnvironmentConfig       *config.EnvironmentConfig
 }
 
 func (m *CSRFMiddleware) Handle(next http.Handler) http.Handler {
+	if m.EnvironmentConfig.End2EndCSRFProtectionDisabled {
+		return next
+	}
```

Comment:
Is it possible that we parse the HTML to get the CSRF token in the HTML so that we do not need to introduce this flag?

Reply:
I think it is possible, but I think csrf in e2e is not very useful so I simply disable it to reduce noises in the tests.

Reply:
It would be useful if we have e2e tests (headless browser-based ones) on authui. Anyway, let's keep this for now.